opentimelineview/timeline_widget.py

```diff
@@ -537,6 +544,186 @@ def wheelEvent(self, event):
+            curTrackYpos = curSelectedItem.parentItem().pos().y()
+
+            newXpos = curItemXpos
+            newYpos = curTrackYpos - TRACK_HEIGHT
+
+            newSelectedItem = self.scene().itemAt(
+                                                  QtCore.QPointF(
+                                                                 newXpos,
+                                                                 newYpos
+                                                                 ),
+                                                  QtGui.QTransform()
+                                                  )
+
+            if not newSelectedItem or isinstance(newSelectedItem, Track):
+                newYpos = newYpos - TRANSITION_HEIGHT
+        else:
+            newXpos = curItemXpos
+            newYpos = curSelectedItem.y()
+
+        newPosition = QtCore.QPointF(newXpos, newYpos)
+
+        return self.scene().itemAt(newPosition, QtGui.QTransform())
+
+    def _get_down_item(self, curSelectedItem):
+        curItemXpos = curSelectedItem.pos().x()
+
+        if curSelectedItem.parentItem():
+            curTrackYpos = curSelectedItem.parentItem().pos().y()
+            newXpos = curItemXpos
+            newYpos = curTrackYpos + TRACK_HEIGHT
+
+            newSelectedItem = self.scene().itemAt(
+                                                  QtCore.QPointF(
+                                                                 newXpos,
+                                                                 newYpos
+                                                                 ),
+                                                  QtGui.QTransform()
+                                                  )
+
+            if not newSelectedItem or isinstance(newSelectedItem, Track):
+                newYpos = newYpos + TRANSITION_HEIGHT
+
+            if newYpos < TRACK_HEIGHT:
+                newYpos = TRACK_HEIGHT
+        else:
+            newXpos = curItemXpos
+            newYpos = MARKER_SIZE + TIME_SLIDER_HEIGHT + 1
+            newYpos = TIME_SLIDER_HEIGHT
+        newPosition = QtCore.QPointF(newXpos, newYpos)
+
+        return self.scene().itemAt(newPosition, QtGui.QTransform())
+
+    def _select_new_item(self, newSelectedItem):
+        # Check for text item
+        # Text item shouldn't be selected,
+        # maybe a bug in the population of timeline.
+        if isinstance(newSelectedItem, QtWidgets.QGraphicsSimpleTextItem):
+            newSelectedItem = newSelectedItem.parentItem()
+
+        # Validate new item for edge cases
+        # If valid, set selected
+        if (
+            not isinstance(newSelectedItem, Track)
+            and newSelectedItem
+        ):
+            if self.scene().selectedItems():
+                for selectedItem in self.scene().selectedItems():
+                    selectedItem.setSelected(False)
+            newSelectedItem.setSelected(True)
+            self.centerOn(newSelectedItem)
```

Comment:
Hi, this is great and helps a lot, however I noticed this centerOn is being called on any keypress (even something like "control"). I wonder if `_get_new_item()` could return `None` for when a new item is not found and only call into `_select_new_item()` if a valid new item was returned. This would limit calls into this `_select_new_item()` method so we could skip the `centerOn()` and `_deselect_all()` code. .

Reply:
Good catch; didn't realize it was it was calling centerOn so frequently. Thanks!

Reply:
You could consider `ensureVisible()` instead of `centerOn()` but in trying it just now, the centering actually works pretty well.